Update to k8s 1.20

surajssd opened the pull request.

surajssd committed:
Vagrantfile: Update VMs to latest Ubuntu

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
etcd: Show the service status

This commit adds a command to show status after the service is enabled
and started so that user can immediately identify if the service came up
fine.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
kube-scheduler: Generate kubeconfig

- Add kube-scheduler CSR.
- Generate kube-scheduler certificate.
- Generate kube-scheduler kubeconfig.
- Modify the kube-scheduler service to include `--kubeconfig` instead of
  `--master` flag.
- Modify automated steps to generate kube-scheduler certs.

This wasn't needed before because scheduler could talk to apiserver on
the localhost insecure port directly. Now that is removed by upstream it
is necessary that all components talk to the apiserver using credentials
of one form or another.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
versions: Update all tools

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
apiserver: Update apiserver health check

Check the health on the, up to date, 6443 (https) port instead of, an
outdated, 8080 (http).

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
controller-services: Show status of services

After enabling and starting services show the status of the services so
user can immediately see that services are running or failing.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
KubeletConfiguration: Use systemd cgroup driver

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
install-tools: Update cfssl download URL

Download from github since it gives more fine grained control on the
version of the cfssl binary.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
download-tools: Don't download unnecessary tools

Now crio ships CNI plugins, runc, conmon and bunch of config.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
etcd: Provide https for localhost

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
apiserver-servie: Add flags for service account signing

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
Vagrantfile: Increase mem of LB machine to 512MB

The lower value of 256MB is not enough to even run `apt install ...`
hence update to the Vagrantfile-level-default of 512MB.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
Vagrantfile: Increase the controller node memory

At 640MB the controller nodes were not performing well, the CPU load
reached double digit to cope up with less memory. Hence increase the
memory for efficient performance of the control plane components.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
cni-config: Update fields according to CRIO

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
coredns: Update the version and config

- Update README command to find the coredns pods.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
setup-workers: Update script

- Install make and conntrack.
- Install crio using make.
- Don't copy lot of unneeded config.
- Show status of services when they are enabled and started.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
crio: Remove unnecessary crio config

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
README: Update the code block tags

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
RBAC: Update from v1beta1 to v1

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
README: Update manual instructions

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

invidian commented on Vagrantfile:
```diff
@@ -2,7 +2,7 @@
 
 Vagrant.configure("2") do |config|
-  config.vm.box = "ubuntu/bionic64"
+  config.vm.box = "ubuntu/hirsute64"
```
I think we should use LTS version where possible.

invidian commented on scripts/setup-etcd:
```diff
@@ -29,5 +29,6 @@ cp \
 systemctl daemon-reload
 systemctl enable etcd
 systemctl start etcd
+systemctl status --no-pager etcd
```
Why is this needed? Please add some explanation in commit message.

invidian commented on certificates/kube-scheduler-csr.json:
```diff
@@ -0,0 +1,16 @@
+{
```
Can you please add some explanation why this was missing before?

invidian commented on scripts/setup-controller-services:
```diff
@@ -44,7 +44,7 @@ systemctl enable kube-apiserver kube-controller-manager kube-scheduler
 EOF
 
-  until vagrant ssh "controller-${i}" -- curl --ipv4 --fail --silent --max-time 3 "http://127.0.0.1:8080/api" &>/dev/null; do
+  until vagrant ssh "controller-${i}" -- curl --ipv4 --fail --silent --max-time 3 -k "https://localhost:6443/healthz" &>/dev/null; do
```
I guess the advantage is to run checks over HTTPS and use more up to date endpoint? Can we mention that in the commit message please?

invidian commented on scripts/setup-controller-services:
```diff
@@ -40,8 +40,10 @@ cp \
 systemctl daemon-reload
-systemctl enable kube-apiserver kube-controller-manager kube-scheduler
-systemctl start kube-apiserver kube-controller-manager kube-scheduler
+services="kube-apiserver kube-controller-manager kube-scheduler"
```
Ditto commit message explanation. I think something like: "... so user can immediately see that services are running (or functional/not crashing, something like this)"

invidian commented on scripts/generate-kubelet-config-file:
```diff
@@ -17,6 +17,7 @@ authentication:
     clientCAFile: "/var/lib/kubernetes/ca.pem"
 authorization:
   mode: Webhook
+cgroupDriver: systemd
```
Is this default with latest Ubuntu? It might not be the case for LTS version.

surajssd replied:
This is set by the runtime provider. That's how we detect what to set for Kubelet by reading the docker's output. What's the benefit of one over other? systemd vs cgroupfs?

invidian replied:
`cgroupfs` is a legacy one (v1) IIRC and `systemd` is v2 (unified). So ideally you would use `systemd`.

surajssd replied:
It worked fine on the LTS version as well.

invidian commented on scripts/install-tools:
```diff
@@ -20,13 +20,13 @@ trap 'popd; rm -rf "${tmpdir}"' EXIT
 source "${dir}/versions.bash"
 
 curl -sSL \
-  -O "https://pkg.cfssl.org/${cfssl_version}/cfssl_${binary_platform}-${binary_format}" \
```
Please add some comment in commit message on why is pkg.cfssl.org no longer a good source.

surajssd replied:
I don't think it is a bad source, but github gives more fine grained control over the version.

invidian replied:
IIRC on GitHub you also have more recent version. Please mention those in commit message :)

surajssd replied:
Yep, that's done.

invidian commented on Vagrantfile:
```diff
@@ -17,7 +17,6 @@ Vagrant.configure("2") do |config|
       c.vm.provision :shell, :path => "scripts/vagrant-setup-haproxy.bash"
 
       c.vm.provider "virtualbox" do |vb|
-        vb.memory = "256"
```
Why?

invidian commented on Vagrantfile:
```diff
@@ -28,7 +28,7 @@ Vagrant.configure("2") do |config|
 
         c.vm.provider "virtualbox" do |vb|
-          vb.memory = "640"
+          vb.memory = "750"
```
750MB still seem a little for controller node. Did you experience some issues with 640MB? Can you please mention that in commit message if so?

invidian commented on manifests/coredns.yaml:
```diff
@@ -75,44 +79,29 @@ metadata:
       - name: coredns
-        image: coredns/coredns:1.6.7
-        imagePullPolicy: IfNotPresent
+        image: quay.io/kinvolk/coredns:1.8.0
```
Why not use `coredns/coredns:1.8.0`?

invidian commented on README.md:
```diff
@@ -168,7 +168,7 @@ Deploy the DNS add-on and verify it's working:
 kubectl apply -f ./manifests/coredns.yaml
 [...]
-kubectl get pods -l k8s-app=kube-dns -n kube-system
+kubectl get pods -l k8s-app=coredns -n kube-system
```
This should probably go with config update commit.

invidian reviewed (commented):
Overall looks good to me. I would be curious about the reasoning behind some of the proposed changes, which is not immediately obvious to me.

surajssd committed:
setup-routes: Install net-tools to workers

Ubuntu 20.04 does not ship net-tools. `route` command is needed for
setting the routes on the host.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

invidian commented on Vagrantfile:
```diff
@@ -2,7 +2,7 @@
 # vi: set ft=ruby :
```
Nit:
```diff
-Vagrantfile: Update VMs to latest ubuntu
+Vagrantfile: Update VMs to latest Ubuntu
```

invidian reviewed (approved):
OK
